org-roam.el

```diff
@@ -251,12 +259,17 @@ it as FILE-PATH."
                (link-type (cond ((and (string= type "file")
                                       (org-roam--org-file-p path))
                                  "roam")
+                                ((string= type "roam")
+                                 "roam")
                                 ((and
                                   (require 'org-ref nil t)
                                   (-contains? org-ref-cite-types type))
                                  "cite")
-                                (t nil))))
-          (when link-type
+                                (t nil)))
+               (is-title (if (string= type "roam") t nil))
+               (roam-file (if is-title (org-roam--get-file-from-title path) t)))
+          ;; Ignore link if it is a roam-link and roam-file doesn't exist
+          (when (and link-type roam-file)
```

Comment:
can be simplified to 

```
(when (and link-type (string= link-type "roam"))
 ...
```

Reply:
This is required to ensure the roam-link points to an existing file, since db enforces 'file' to be non-null:
`(roam-file (if is-title (org-roam--get-file-from-title path) t)))`

`(is-title (if (string= type "roam") t nil))` was partly to reduce verbosity, and line 288 uses it to differentiate filename processing between roam- and file-links (which are both passed as link-type "roam").

Reply:
Tried to simplify by introducing "org" type. Updated org-roam-db in two places to work with the change. Don't think it affects anything else?